sdk/src/main/java/com/hashicorp/nomad/javasdk/NomadPredicates.java

```diff
@@ -112,4 +113,86 @@ public boolean apply(T value) {
             }
         };
     }
+
+    /**
+     * Returns a predicate that is true when either of the given predicates is true.
+     */
+    public static <T> Predicate<T> either(final Predicate<? super T> a, final Predicate<? super T> b) {
+        return new Predicate<T>() {
+            @Override
+            public boolean apply(T value) {
+                return a.apply(value) || b.apply(value);
+            }
+        };
+    }
+
+    /**
+     * Returns a predicate that checks if allocation has the given client status.
+     */
+    public static Predicate<AllocationListStub> allocationHasClientStatus(final String status) {
+        return new Predicate<AllocationListStub>() {
+            @Override
+            public boolean apply(@Nonnull AllocationListStub allocationListStub) {
+                return status.equals(allocationListStub.getClientStatus());
+            }
+        };
+    }
+
+    /**
+     * Returns a predicate that checks if allocation has completed.
+     */
+    public static Predicate<AllocationListStub> allocationHasCompleted() {
+        return allocationHasClientStatus("complete");
+    }
+
+    /**
+     * Returns a predicate that checks if allocation has failed.
+     */
+    public static Predicate<AllocationListStub> allocationHasFailed() {
+        return allocationHasClientStatus("failed");
+    }
+
+    /**
+     * Returns a predicate that checks if allocation has completed successfully or failed (i.e. is not in progress).
+     */
+    public static Predicate<AllocationListStub> allocationFinishedRunning() {
+        return either(allocationHasCompleted(), allocationHasFailed());
+    }
+
+    /**
+     * Returns a predicate that checks if all allocations from a given list have finished running.
+     */
+    public static Predicate<List<AllocationListStub>> allAllocationsFinished() {
+        return new Predicate<List<AllocationListStub>>() {
+            @Override
+            public boolean apply(List<AllocationListStub> allocs) {
+                for (AllocationListStub alloc : allocs) {
+                    if (!allocationFinishedRunning().apply(alloc)) {
+                        return false;
+                    }
+                }
+                return true;
+            }
+        };
+    }
+
+    /**
+     * Returns a predicate that checks if more than a given percentage of allocations finished with status "failed".
+     */
+    public static Predicate<List<AllocationListStub>> failedAllocationsOver(final Long threshold) {
```

Comment:
This seems rather specific to a particular use case. Maybe it should be implemented by caller?